### PR TITLE
Remove getByBlockNumber from ProtocolScheduleBuilder

### PR DESCRIPTION
**Pull request by gfukushima**
## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Acceptance Tests (Non Mainnet)

- [ ] I have considered running `./gradlew acceptanceTestNonMainnet` locally if my PR affects non-mainnet modules.

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ProtocolScheduleBuilder.java [hunk 1]
```diff
@@ -192,7 +192,12 @@ protected void postBuildStep(final MainnetProtocolSpecFactory specFactory) {
         .ifPresent(
             daoBlockNumber -> {
               final ProtocolSpec originalProtocolSpec =
-                  protocolSchedule.getByBlockNumber(daoBlockNumber);
+                  protocolSchedule.getScheduledProtocolSpecs().stream()
+                      .filter(x -> x.getBlock() <= daoBlockNumber)
```
Comment (siladu) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ProtocolScheduleBuilder.java:196]: I think this approach is acceptable, but it does have some minor issues. It technically breaks encapsulation so someone might attempt something like this...
```
 blockAddedEvent -> {
   if (protocolSchedule.getScheduledProtocolSpecs().stream()
      .anyMatch(s ->
         s.milestone() == blockAddedEvent.getBlock().getHeader().getTimestamp()
         || s.milestone() == blockAddedEvent.getBlock().getHeader().getNumber())) {
```

which leaks the complexity of numbers vs timestamp outside of the schedule.

We could limit the visibility of getScheduledProtocolSpecs() to be in the same package (although would have to deal with CombinedProtocolSchedule).
We could also leave it only implemented in MutableProtocolSchedule, but that will cause problems when we want to unify the schedules.

Comment (gfukushima) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ProtocolScheduleBuilder.java:196]: Closing this PR in favor of #5159

**Comment by gfukushima** [comment]
Closing this PR in favor of https://github.com/hyperledger/besu/issues/5159

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ProtocolScheduleBuilder.java (hunk 1 above)
Comment (jframe) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ProtocolScheduleBuilder.java:196]: I like the simplicity of this, but I don't think it's the right way to do this. As was said above we just reimplementing the getByBlockNumber functionality outside of the protocol schedule.

Comment (gfukushima) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ProtocolScheduleBuilder.java:196]: This PR has already been closed in favor of another PR, thanks though!

**Pull request closed without merging**